Create documentation for Omega Walls integration (#462)
deepset-ai/haystack-integrations: integrations/omega-walls.md

---
layout: integration
name: Omega Walls

> description: Runtime trust boundary for Haystack RAG and agent pipelines that ingest untrusted content.

Reviewer: ```suggestion
description: Runtime trust boundary for ingested content in Haystack applications
```

Omega Walls adds a boundary between:

- untrusted content
- model context
- memory
- tools and actions

The Haystack adapter provides a guard layer that can wrap a Haystack pipeline and tool calls before execution.

## Installation

Install Omega Walls with framework integrations:

```bash
pip install "omega-walls[integrations]"
````

## Usage

### Components

This integration introduces the `OmegaHaystackGuard`, which can be used to wrap a Haystack pipeline and guard tool calls.

* `OmegaHaystackGuard.wrap_pipeline(...)`: wraps a Haystack pipeline with an Omega guard component.
* `OmegaHaystackGuard.wrap_tool(...)`: wraps tool functions so tool calls can be checked before execution.

### Use Omega Walls with a Haystack pipeline

```python
from omega.integrations import OmegaHaystackGuard

guard = OmegaHaystackGuard(profile="quickstart")

pipeline = guard.wrap_pipeline(
    pipeline,
    component_name="omega_guard_component",
)

> safe_tool = guard.wrap_tool("network_post", network_post_fn)

Reviewer: Can I then use the pipeline/the tool as usual? What's the process after this step?

```

### What it checks

Omega Walls is designed to check:

* model/input flow
* tool calls before execution
* memory-write candidates
* cross-document or cross-step prompt-injection pressure
* secret-exfiltration pressure
* tool/action abuse

When a risky input or tool call is detected, the guard can block the step with typed exceptions instead of letting the pipeline continue blindly.

## When to use it

Omega Walls is useful for Haystack users building RAG or agentic workflows over real-world content, especially when the pipeline reads from:

* external documents
* PDFs
* web pages
* support tickets
* user-provided files
* internal knowledge bases
* tool outputs

Typical use cases include:

* internal document assistants
* support-ticket summarizers
* agentic RAG workflows
* pipelines with tool calls

> * pipelines that read untrusted or user-provided documents

Reviewer: Can you shorten this section in couple of sentences? Otherwise, it takes too much space
